__tests__/addSerializers.test.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+const React = require('react');
+const { configure, shallow } = require('enzyme');
+const Adapter = require('enzyme-adapter-react-16');
+const enzymeToJson = require('enzyme-to-json/serializer');
+const snapshotDiff = require('../src/index');
+
+configure({ adapter: new Adapter() });
+snapshotDiff.addSerializers([enzymeToJson]);
+
+type Props = {
+  test: string,
+};
+
+class Component extends React.Component<Props> {
+  render() {
+    return (
+      <div>
+        <span />
+        <span />
+        <span />
+        <span>{this.props.test}</span>
+        <span>{this.props.test}</span>
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span>{this.props.test}</span>
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+        <span />
+      </div>
+    );
+  }
+}
+
+test('detects enzyme shallow rendered components', () => {
+  expect(
+    snapshotDiff(
+      shallow(<Component test="say" />),
+      shallow(<Component test="my name" />)
+    )
+  ).toMatchSnapshot();
+});
+
+test('can use contextLines with enzyme shallow rendered components', () => {
+  expect(
+    snapshotDiff(
+      shallow(<Component test="say" />),
+      shallow(<Component test="my name" />),
+      {
+        contextLines: 0,
+      }
+    )
+  ).toMatchSnapshot();
+});
```

Comment:
Tests are a copy of those used for the react component testing in `snapshotDiff.test.js`